### PR TITLE
Update sidebar & breadcrumbs when editing group

### PR DESCRIPTION
**Pull request by ksierks**
## What does this PR do and why?
Updates the group name in the layout sidebar and breadcrumbs when a group name is successfully updated.
Pulled from [#192](https://github.com/phac-nml/irida-next/pull/192).

## Screenshots or screen recordings
Screenshot from 2023-09-25 14-41-59

## How to set up and validate locally
1. Navigate to a Group
2. Click the `Settings` link in the sidebar
3. In the `Group name and description` form set `Group Name` to `a`
4. Click `Update group`
5. See that the form displays an error message and the sidebar and breadcrumbs are not updated.
6. Change the `Group Name` to `New group name`.
7. Click `Update group`
8. See a success message and that the sidebar an breadcrumbs have changed to the new name.

Note: Verify that change path refreshes the page to the new URL as expected.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: test/controllers/groups_controller_test.rb [hunk 1]
```diff
@@ -51,16 +51,16 @@ class GroupsControllerTest < ActionDispatch::IntegrationTest # rubocop:disable M
     sign_in users(:john_doe)
 
     group = groups(:group_one)
-    patch group_path(group), params: { group: { name: 'New Group Name' } }
-    assert_redirected_to group_path(group)
+    patch group_path(group, format: :turbo_stream), params: { group: { name: 'New Group Name' } }
```
Comment (deepsidhu85) [inline test/controllers/groups_controller_test.rb:54]: We should move the `format: :turbo_stream` to the params hash before the closing curly brace as we did for the projects_controller_test. Just to keep it consistent

File: test/controllers/groups_controller_test.rb [hunk 2]
```diff
@@ -51,16 +51,16 @@ class GroupsControllerTest < ActionDispatch::IntegrationTest # rubocop:disable M
...
+    assert_response :ok
   end
 
   test 'should not update a group with invalid params' do
     sign_in users(:john_doe)
 
     group = groups(:group_one)
     assert_no_changes -> { group.name } do
-      patch group_path(group), params: { group: { name: 'NG' } }
+      patch group_path(group, format: :turbo_stream), params: { group: { name: 'NG' } }
```
Comment (deepsidhu85) [inline test/controllers/groups_controller_test.rb:63]: Same comment as above

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This looks good to me! Just a couple of changes below

**Review by joshsadam — APPROVED** [review]
UI updates as I expected.  This should be good to go after making @deepsidhu85 's changes.

File: test/controllers/groups_controller_test.rb (hunk 1 above)
Comment (ksierks) [reply test/controllers/groups_controller_test.rb]: Please see e85abc419e4970d1f29684887d4744e5f8f1a58e.

File: test/controllers/groups_controller_test.rb (hunk 2 above)
Comment (ksierks) [reply test/controllers/groups_controller_test.rb]: Please see e85abc419e4970d1f29684887d4744e5f8f1a58e.

**Review by joshsadam — APPROVED** [review]
Thanks for making those changes the @deepsidhu85 requested.  This looks good now and works great.

**Review by joshsadam — APPROVED** [review]
Thanks for making those changes the @deepsidhu85 requested.  This looks good now and works great.

**Pull request merged**